### PR TITLE
update: Calico app policy  - Isito 1.15.x 

### PR DESCRIPTION
**Pull request by frozenprocess**
## Description
Since Istio 1.10.x reached its EOL (Jan 7, 2022) I thought we need to add support for the latest version.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Update Istio support to include Istio v1.15.2
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: manifests/alp/istio-inject-configmap-1.15.yaml [hunk 1]
```diff
@@ -0,0 +1,1199 @@
...
+              name: dikastes-sock
+
+          - name: dikastes
+            image: calico/dikastes:master
+            args: ["server", "-l", "/var/run/dikastes/dikastes.sock", "-d", "/var/run/felix/nodeagent/socket"]
+            securityContext:
+              allowPrivilegeEscalation: false
+              runAsGroup: 0
+              runAsNonRoot: false
+              runAsUser: 0
```
Comment (frozenprocess) [inline manifests/alp/istio-inject-configmap-1.15.yaml:443]: AKS, and EKS PSP problem, please doublecheck these permissions before merging.
Thanq!

**Comment by caseydavenport** [comment]
@mgleung could you take a look at this one?

**Review by mgleung — APPROVED** [review]

File: manifests/alp/istio-inject-configmap-1.15.yaml (hunk 1 above)
Comment (mgleung) [reply manifests/alp/istio-inject-configmap-1.15.yaml:443]: I may have just approved this pre-emptively :stuck_out_tongue: but have we tested if this is needed for the dikastes container? I remember this being unnecessary last time I checked.

**Review by mgleung — CHANGES_REQUESTED** [review]

File: manifests/alp/istio-inject-configmap-1.15.yaml (hunk 1 above)
Comment (frozenprocess) [reply manifests/alp/istio-inject-configmap-1.15.yaml:443]: pre-emptively 😈
In a local cluster, it works without these permissions, but in AKS `dikastes` prompts the "Unable to set write permission on socket." error message. 
https://github.com/projectcalico/calico/blob/baa0624e050dd8ca29a53148cfa32d37838640dd/app-policy/cmd/dikastes/dikastes.go#L103-L106

File: manifests/alp/istio-inject-configmap-1.15.yaml [hunk 2]
```diff
@@ -0,0 +1,1199 @@
...
+              {{ end }}
+              {{- end }}
+            - mountPath: /var/run/dikastes
+              name: dikastes-sock
+
+          - name: dikastes
+            image: calico/dikastes:master
+            args: ["server", "-l", "/var/run/dikastes/dikastes.sock", "-d", "/var/run/felix/nodeagent/socket"]
+            securityContext:
+              allowPrivilegeEscalation: false
```
Comment (frozenprocess) [inline manifests/alp/istio-inject-configmap-1.15.yaml:440]: I've changed the permission to use the same user as `dikastes` 
https://github.com/projectcalico/calico/blob/9d890ab5963a0bededc110cf687c2287f225d843/manifests/alp/istio-inject-configmap-1.15.yaml#L436-L443

https://github.com/projectcalico/calico/blob/c69d0874f45ef2d5ef7cff53bcbe9d4416d71b14/app-policy/Dockerfile.amd64#L57


@mgleung WDYT?

Comment (mgleung) [reply manifests/alp/istio-inject-configmap-1.15.yaml:440]: This is awesome. Thanks for making that change.

**Review by mgleung — APPROVED** [review]

**Pull request merged**